boards/FSM/Src/App/App_Signals.c

```diff
@@ -0,0 +1,52 @@
+#include "App_AcceleratorPedal.h"
+#include "App_Signals.h"
+
+struct AcceleratorPedal *_papps;
+struct AcceleratorPedal *_sapps;
+
+static bool IsPappsAlarmActive(void)
+{
+    return App_AcceleratorPedal_IsEncoderAlarmActive(_papps);
+}
+
+static void PappsAlarmCallback(void *context)
+{
+    struct FsmWorld *         world  = (struct FsmWorld *)context;
+    struct FsmCanTxInterface *can_tx = App_FsmWorld_GetCanTx(world);
+
+    App_CanTx_SetPeriodicSignal_PAPPS_MAPPED_PEDAL_PERCENTAGE(can_tx, 0.0f);
+}
+
+static bool IsSappsAlarmActive(void)
+{
+    return App_AcceleratorPedal_IsEncoderAlarmActive(_papps);
+}
+
+static void SappsAlarmCallback(void *context)
```

Comment:
I think I'd prefer we just pass in either `World` or `SharedStateMachine` here, just because it's more clear.

Reply:
This function signature has to conform to `struct SignalCallback` so it must be a void pointer

```
    8   struct SignalCallback                                                                                                                                                                                                                             
    1 {                                                                                                                                                                                                                                                 
    2     // How long the signal must be continuously high for, in milliseconds,                                                                                                                                                                        
    3     // before the callback function is triggered                                                                                                                                                                                                  
    4     uint32_t high_duration_ms;                                                                                                                                                                                                                    
    5                                                                                                                                                                                                                                                   
    6     // The callback function                                                                                                                                                                                                                      
    7     void (*function)(void *);                                                                                                                                                                                                                     
    8                                                                                                                                                                                                                                                   
    9     // The context to pass into the callback function. Set as NULL if unused.                                                                                                                                                                     
   10     void *context;                                                                                                                                                                                                                                
   11 };                                                                                                                                                                                                                                                
   12    
```
I can't think of an elegant way to do this otherwise but let me if there's a clever solution I'm missing

Reply:
Why not have function just take a `World` instead?

Reply:
I guess as of now `Signal` is meant to be used in conjunction with `World*` only so this would make sense. Changed!